pkgs/development/ocaml-modules/bap/default.nix

```diff
@@ -0,0 +1,68 @@
+{stdenv, buildOcaml, fetchFromGitHub, fetchurl, camlp4, ocaml_oasis, bitstring, camlzip, cmdliner, core_kernel, ezjsonm, faillib, fileutils, ocaml_lwt, ocamlgraph, ocurl, re, uri, zarith, piqi, piqi-ocaml, uuidm, llvm_38, ulex, easy-format, xmlm, frontc, ounit, utop, which, makeWrapper, writeText, ocaml}:
+
+buildOcaml rec {
+  name = "bap";
+  version = "1.2.0";
+  src = fetchFromGitHub {
+    owner = "BinaryAnalysisPlatform";
+    repo = "bap";
+    rev = "v${version}";
+    sha256 = "0dn1gvj73pma0rsw8r50cmjddibnf42w1kbskb2vpzq0kb79jlkw";
+  };
+
+  sigs = fetchurl {
+     url = "https://github.com/BinaryAnalysisPlatform/bap/releases/download/v${version}/sigs.zip";
+     sha256 = "0mpsq2pinbrynlisnh8j3nrlamlsls7lza0bkqnm9szqjjdmcgfn";
+  };
+
+  createFindlibDestdir = true;
+
+  setupHook = writeText "setupHook.sh" ''
+    export CAML_LD_LIBRARY_PATH="''${CAML_LD_LIBRARY_PATH}''${CAML_LD_LIBRARY_PATH:+:}''$1/lib/ocaml/${ocaml.version}/site-lib/${name}/"
+    export CAML_LD_LIBRARY_PATH="''${CAML_LD_LIBRARY_PATH}''${CAML_LD_LIBRARY_PATH:+:}''$1/lib/ocaml/${ocaml.version}/site-lib/${name}-llvm-plugins/"
+  '';
+
+  buildInputs = [ ocaml_oasis
+                  llvm_38
+                  utop
+                  #build tricks
+                  which
+                  makeWrapper
+                  ];
+
+  propagatedBuildInputs = [ bitstring camlzip cmdliner core_kernel ezjsonm faillib fileutils ocaml_lwt ocamlgraph ocurl re uri zarith piqi
+                            piqi-ocaml uuidm frontc ounit ];
+
+  installPhase = ''
+  export OCAMLPATH=$OCAMLPATH:$OCAMLFIND_DESTDIR;
+  export PATH=$PATH:$out/bin
+  export CAML_LD_LIBRARY_PATH=$CAML_LD_LIBRARY_PATH:$OCAMLFIND_DESTDIR/bap-plugin-llvm/:$OCAMLFIND_DESTDIR/bap/
+  cat <<EOF > tools/baptop
+  #!/bin/bash
+  export OCAMLPATH=\$OCAMLPATH:`echo $OCAMLPATH`
+  export PATH=\$PATH:`echo $PATH`
+  export CAML_LD_LIBRARY_PATH=\$CAML_LD_LIBRARY_PATH:`echo $CAML_LD_LIBRARY_PATH`
+  exec ${utop}/bin/utop -ppx ppx-bap -short-paths -require "bap.top" "\$@"
+  EOF
+  mkdir -p $out/lib/bap
+  make install
+  wrapProgram $out/bin/bapbuild --set OCAMLPATH `echo $OCAMLPATH` --set PATH `echo $PATH`
```

Comment:
Excluding external OCAMLPATH and PATH is intended here? (`--set` instead of `--prefix`).